Complate task for lesson1

yaNimotsi opened the pull request:
Правильна ли была задумка сначала выполнить все запросы, получить ответы и только потом, общим string, записывать в txt?

Commit:
Complate task for lesson1

MeatBatyr commented on Lesson1/Program.cs:
```diff
@@ -1,12 +1,126 @@
+        {
+            try
+            {
+                var responseQuery = Program.Client.GetAsync(postAddressUri, HttpCompletionOption.ResponseContentRead, Cts.Token).Result;
```
await , не успользуй .Result в асинхронных функциях

yaNimotsi replied:
> await , не успользуй .Result в асинхронных функциях

Внес корректировки по данному замечанию

MeatBatyr commented on Lesson1/Program.cs:
```diff
@@ -1,12 +1,126 @@
+    internal static class Program
     {
-        static void Main(string[] args)
+        private static readonly HttpClient Client = new();
```
нельзя хранить в статическом поле класс, который реализует IDIsposable

yaNimotsi replied:
исправил

yaNimotsi replied:
> нельзя хранить в статическом поле класс, который реализует IDIsposable

Внес корректировки в код

Commit:
Move statick field Client (HTTP Client) to Main method. ReName Client to client

Commit:
Update logick get response form the post

MeatBatyr commented on Lesson1/Program.cs:
```diff
@@ -1,12 +1,139 @@
+
+            var result = new List<string>();
+
+            for (int i = startPostIndex; i <= endPostIndex; i++)
```
Идея была в параллельном запросе всех постов, а тут в цикле последовательно выполняются все.
Через цикл нужно было создать таски, и их выполнить через WaitAll/WhenAll

yaNimotsi replied:
> Идея была в параллельном запросе всех постов, а тут в цикле последовательно выполняются все. Через цикл нужно было создать таски, и их выполнить через WaitAll/WhenAll

Исправил. Кажется понял свою ошибку

Commit:
Исправил согласно последним замечаниям.
Создание Task перенесено в Main. Ожидается их выполнение и потом приступается к синхронному выполнению записи в текстовый документ.
